src/nvim/event/process.c

```diff
@@ -256,14 +242,17 @@ static void children_kill_cb(uv_timer_t *handle)
       continue;
     }
     uint64_t term_sent = UINT64_MAX == proc->stopped_time;
-    if (kProcessTypePty != proc->type || term_sent) {
-      os_proc_tree_kill(proc->pid, SIGKILL);
-    } else {
-      os_proc_tree_kill(proc->pid, SIGTERM);
-      proc->stopped_time = UINT64_MAX;  // Flag: SIGTERM was sent.
+    if (proc->type == kProcessTypePty && !term_sent) {
+      // close all streams for pty processes to send SIGHUP to the process
+      process_close_streams(proc);
+      pty_process_close_master((PtyProcess *)proc);
+
+      proc->stopped_time = UINT64_MAX;  // Flag: SIGHUP was sent.
       // Restart timer.
       uv_timer_start(&proc->loop->children_kill_timer, children_kill_cb,
                      KILL_TIMEOUT_MS, 0);
+    } else {
+      os_proc_tree_kill(proc->pid, SIGKILL);
```

Comment:
this "smaller" case was intentionally on top, so that it is easier to visually scan the if/else cases.

Reply:
I see.
I found it easier to see the first step first.. ;) - i.e. SIGKILL will come after.
Can be reverted however.